### PR TITLE
[TRUS-757] [TRUS-1011] - Summary Answers & Confirmation Answers print pages

### PR DESCRIPTION
**Pull request by jassalrichy**
[TRUS-757]  Summary check page - Print or Save
[TRUS-1011]  Printing from the Confirmation Page

### CODE REVIEW

File: app/controllers/SummaryAnswerPageController.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      val checkYourAnswersHelper = new CheckYourAnswersHelper(countryOptions)(request.userAnswers, isSummary = true)
+      val trustDetails = checkYourAnswersHelper.trustDetails.getOrElse(Nil)
+      val trustees = checkYourAnswersHelper.trustees.getOrElse(Nil)
+      val settlors = checkYourAnswersHelper.settlors.getOrElse(Nil)
+      val individualBeneficiaries = checkYourAnswersHelper.individualBeneficiaries.getOrElse(Nil)
+      val classOfBeneficiaries = checkYourAnswersHelper.classOfBeneficiaries.getOrElse(Nil)
+      val moneyAsset = checkYourAnswersHelper.moneyAsset.getOrElse(Nil)
+      val sections =  Seq() ++ trustDetails ++ settlors ++ trustees ++ individualBeneficiaries ++ classOfBeneficiaries ++ moneyAsset
+
+      val isTaskListComplete = registrationProgress.isTaskListComplete(request.userAnswers)
```
Comment (manishsw) [inline app/controllers/SummaryAnswerPageController.scala]: @jassalrichy  RegistrationComplete refiner available to check the task list complete, check declaration controller.

File: app/controllers/ConfirmationAnswerPageController.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      val classOfBeneficiaries = checkYourAnswersHelper.classOfBeneficiaries.getOrElse(Nil)
+      val moneyAsset = checkYourAnswersHelper.moneyAsset.getOrElse(Nil)
+      val sections =  Seq() ++ trustDetails ++ settlors ++ trustees ++ individualBeneficiaries ++ classOfBeneficiaries ++ moneyAsset
+
+      val isTaskListComplete = registrationProgress.isTaskListComplete(request.userAnswers)
+
+      val trn = request.userAnswers.get(RegistrationTRNPage).getOrElse("")
+      val trnDateTime = request.userAnswers.get(RegistrationSubmissionDatePage).getOrElse(LocalDateTime.now)
+
+      val dateFormatter = DateTimeFormatter.ofPattern("d MMMM yyyy")
```
Comment (manishsw) [inline app/controllers/ConfirmationAnswerPageController.scala]: @jassalrichy move this to class level

File: app/views/components/answer_section.scala.html [hunk 3]
```diff
@@ -19,6 +19,10 @@
 
 @(answerSection: AnswerSection)(implicit messages: Messages)
 
+@if(answerSection.sectionKey.isDefined){
+    <h2 class="heading-large">@messages(answerSection.sectionKey.get)</h2>
```
Comment (adamconder) [inline app/views/components/answer_section.scala.html]: Is it safe to call `.get`. I don't think so?

File: app/views/components/link.scala.html [hunk 4]
```diff
@@ -14,6 +14,6 @@
  * limitations under the License.
  *@
 
-@(href : String, id : String = "", key : String, target : String = "")(implicit messages: Messages)
+@(href : String, id : String = "", key : String, openInNewWindow: Boolean = false)(implicit messages: Messages)
 
-<a id="@id" target="@target" href="@href">@messages(key)</a>
+<a id="@id" href="@href" @if(openInNewWindow) {target="_blank"}>@messages(key)</a>
```
Comment (adamconder) [inline app/views/components/link.scala.html]: 👍 Thanks!

File: conf/app.routes [hunk 5]
```diff
@@ -371,4 +371,8 @@ POST       /change-agent-international-address                  controllers.Agen
 GET        /declaration                        controllers.DeclarationController.onPageLoad(mode: Mode = NormalMode)
 POST       /declaration                        controllers.DeclarationController.onSubmit(mode: Mode = NormalMode)
 
-GET       /confirmation                         controllers.ConfirmationController.onPageLoad()
\ No newline at end of file
+GET       /confirmation                         controllers.ConfirmationController.onPageLoad()
+
+GET        /your-answers-summary                          controllers.SummaryAnswerPageController.onPageLoad()
```
Comment (adamconder) [inline conf/app.routes]: Anita has asked if these could become `/your-draft-registration` and `/your-declared-registration`.

File: test/views/ConfirmationAnswerPageViewSpec.scala [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+class ConfirmationAnswerPageViewSpec extends ViewBehaviours {
+  val index = 0
+
+  "ConfirmationAnswerPage view" must {
+
+    val view = viewFor[ConfirmationAnswerPageView](Some(emptyUserAnswers))
+
+    val applyView = view.apply(Nil, "", "")(fakeRequest, messages)
+
+    behave like normalPage(applyView, "confirmationAnswerPage")
```
Comment (adamconder) [inline test/views/ConfirmationAnswerPageViewSpec.scala]: Do we want to add any view specs as we're rendering a lot of information? Worth having a conversation with @JohnWhitfield to understand how much of this is done between unit tests and journey tests.

File: test/views/SummaryAnswerPageViewSpec.scala [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+class SummaryAnswerPageViewSpec extends ViewBehaviours {
+  val index = 0
+
+  "SummaryAnswerPage view" must {
+
+    val view = viewFor[SummaryAnswerPageView](Some(emptyUserAnswers))
+
+    val applyView = view.apply(Nil)(fakeRequest, messages)
+
+    behave like normalPage(applyView, "summaryAnswerPage")
```
Comment (adamconder) [inline test/views/SummaryAnswerPageViewSpec.scala]: Same comment as above to have a conversation with @JohnWhitfield

File: app/utils/CheckYourAnswersHelper.scala [hunk 8]
```diff
@@ -20,24 +20,156 @@ import java.time.format.DateTimeFormatter
...
 import pages._
 import play.api.i18n.Messages
 import play.twirl.api.{Html, HtmlFormat}
 import uk.gov.hmrc.domain.Nino
 import utils.CheckYourAnswersHelper.{indBeneficiaryName, trusteeName, _}
 import utils.countryOptions.CountryOptions
-import viewmodels.AnswerRow
+import viewmodels.{AnswerRow, AnswerSection}
+
+class CheckYourAnswersHelper @Inject()(countryOptions: CountryOptions)(userAnswers: UserAnswers, isSummary: Boolean = false)(implicit messages: Messages)  {
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/utils/CheckYourAnswersHelper.scala]: Done

File: app/utils/CheckYourAnswersHelper.scala [hunk 9]
```diff
@@ -20,24 +20,156 @@ import java.time.format.DateTimeFormatter
...
+    if (questions.nonEmpty) Some(Seq(AnswerSection(None, questions, Some(Messages("answerPage.section.settlors.heading"))))) else None
+  }
+
+  def trustees : Option[Seq[AnswerSection]] = {
+    for {
+      trustees <- userAnswers.get(Trustees)
+      indexed = trustees.zipWithIndex
+    } yield indexed.map {
+      case (trustee, index) =>
+        val isLead = userAnswers.get(IsThisLeadTrusteePage(index)).get
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/utils/CheckYourAnswersHelper.scala]: Done

File: app/controllers/ConfirmationAnswerPageController.scala (hunk 2 above)
Comment (jassalrichy) [reply app/controllers/ConfirmationAnswerPageController.scala]: Done

File: app/controllers/SummaryAnswerPageController.scala (hunk 1 above)
Comment (jassalrichy) [reply app/controllers/SummaryAnswerPageController.scala]: Done

File: app/controllers/ConfirmationAnswerPageController.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/ConfirmationAnswerPageController.scala]: Done

File: app/controllers/SummaryAnswerPageController.scala [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    implicit request =>
+
+      val checkYourAnswersHelper = new CheckYourAnswersHelper(countryOptions)(request.userAnswers, isSummary = true)
+      val trustDetails = checkYourAnswersHelper.trustDetails.getOrElse(Nil)
+      val trustees = checkYourAnswersHelper.trustees.getOrElse(Nil)
+      val settlors = checkYourAnswersHelper.settlors.getOrElse(Nil)
+      val individualBeneficiaries = checkYourAnswersHelper.individualBeneficiaries.getOrElse(Nil)
+      val classOfBeneficiaries = checkYourAnswersHelper.classOfBeneficiaries.getOrElse(Nil)
+      val moneyAsset = checkYourAnswersHelper.moneyAsset.getOrElse(Nil)
+      val sections =  Seq() ++ trustDetails ++ settlors ++ trustees ++ individualBeneficiaries ++ classOfBeneficiaries ++ moneyAsset
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/SummaryAnswerPageController.scala]: Done

File: app/views/components/answer_section.scala.html (hunk 3 above)
Comment (jassalrichy) [reply app/views/components/answer_section.scala.html]: the previous line checks that the sectionKey cannot be None ?

File: conf/app.routes (hunk 5 above)
Comment (jassalrichy) [reply conf/app.routes]: Done

File: app/utils/CheckYourAnswersHelper.scala [hunk 11]
```diff
@@ -20,24 +20,156 @@ import java.time.format.DateTimeFormatter
...
+          trusteesUkAddress(index),
+          telephoneNumber(index)
+        ).flatten
+
+        AnswerSection(Some(Messages("answerPage.section.trustee.subheading") + " " + (index + 1)),
+          questions, if (index == 0) Some(Messages("answerPage.section.trustees.heading")) else None)
+    }
+  }
+
+  private var displaySectionHeading: Boolean = true
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/utils/CheckYourAnswersHelper.scala]: Done

File: test/views/ConfirmationAnswerPageViewSpec.scala (hunk 6 above)
Comment (jassalrichy) [reply test/views/ConfirmationAnswerPageViewSpec.scala]: Done

File: test/views/SummaryAnswerPageViewSpec.scala (hunk 7 above)
Comment (jassalrichy) [reply test/views/SummaryAnswerPageViewSpec.scala]: Done

File: app/views/components/answer_section.scala.html (hunk 3 above)
Comment (adamconder) [reply app/views/components/answer_section.scala.html]: That's fine 👍

File: app/assets/stylesheets/trustsfrontend-app.scss [hunk 12]
```diff
@@ -166,4 +166,17 @@
...
+}
+
+.declarationReferenceNumber {font-weight:700;}
+
+@media print {
+    .print-hidden, .header-wrapper, .service-info, .report-error  {
+        display: none !important;
+    }
+    .content__body {width: 100%;}
+    .heading-xlarge {font-size: 27px;}
```
Comment (adamconder) [inline app/assets/stylesheets/trustsfrontend-app.scss]: Adam to check if fixed size works on mobile.

**Comment by adamconder** [comment]
👍 Happy with the changes.

**Comment by jassalrichy** [comment]
@adamconder updated with agent client reference on summary answers page

**Pull request merged**